### ISSUE TITLE
Sync of mute button with the application

### ISSUE THREAD

**Issue by vivekkalyanarangan30** [jimstump/trinket-m0-physical-mute-button-for-video-chat#1]
Hi Team! Fantastic project! I am looking to extend this further so that the mute button is always in sync with the application.

Scenario: I mute the Zoom call with the physical mute button --> Use my mouse to unmute --> In this case the LED on the physical device should change to refresh the latest status. How can this be achieved?

**Comment by jimstump** [comment]
I'm very sorry for the late reply.  I never got the notification that you raised this issue.

As mentioned in the [TODO section of the Readme](https://github.com/jimstump/trinket-m0-physical-mute-button-for-video-chat#todo), syncing the state of the button is something that I've thought about and briefly looked into.  Since this device currently attaches to the computer as a keyboard (and then just sends the standard keyboard shortcuts for the appropriate action) I don't know of a way to read the current mute state of the application without writing and installing some additional software on the host machine.

I did link to a [different project](https://github.com/toelke/google-hangouts-muter) where someone wrote some Python code that uses Windows APIs to grab the section of the screen in Google Hangouts where the mute indicator would appear to determine the mute state.  It _should_ be possible to adapt this to Zoom and other video chat applications on Windows (not sure about other operating systems), but would be brittle to design changes made by the individual applications.


I originally cobbled this solution together to help my Kindergarten daughter navigate virtual learning without 100% attention from me, and in that regard it worked pretty well.  She only used the hardware buttons and didn't multitask (where the wrong window/application would then receive the keyboard shortcut).  She was able to easily unmute to answer questions and mute herself again when she was done.  When the class took a break for snack time, she was also able to turn off her camera without assistance.

An original plan was to get this project into a state where I could produce enough buttons to give to the rest of the students in her class to help everyone else too.  For that to work, I decided that it needed to be "plug and play", support as many of the various platforms that people were using for virtual learning as possible, and not require any additional support from me (this precluded custom software running on the machine).  Unfortunately, I found that plan to be cost-prohibitive.  It wasn't "cheap" enough to give them away, and honestly the novelty of hardware mute buttons didn't seem to be greater than the sum of its components.  I didn't expect someone to pay what it cost me in parts to make the buttons, let alone account for my time.  So instead, I just ended up building a couple of prototypes and that was it.


Long story short, even so many months later, I'm not sure how I could portably and reliably read the mute state of the video chat application from the Arduino.  I'm open to ideas (and pull requests) though.

**Comment by vivekkalyanarangan30** [comment]
Hey @jimstump I am sorry for the late reply as well, never got around to checking my notifs.
I achieved this a few months ago with pyautogui and screengrab. I had to account for variations like dark mode, position of the mute button, but with a few if-else statements it fairly works well now for zoom, teams and gmeet. Thanks for your help!

**Comment by jimstump** [comment]
@vivekkalyanarangan30 Can you share that code?  I’d be interested in how that all worked.
